src/pages/Location/Location.js

```diff
@@ -57,8 +81,54 @@ const LocationComponent = () => {
   const [selectedToEditSite, setSelectedToEditSite] = useState({});
   const [handleConfirm, setHandleConfirm] = useState(false);
   const [locZipCode, setLocZipCode] = useState();
+  const [currArea, setCurrArea] = useState([]);
+  const [fieldDialogState, setFieldDialogState] = useState(initFieldDialogState);
+  const [selectedUserField, setSelectedUserField] = useState(
+    userFieldRedux && userFieldRedux.data.length
+      ? userFieldRedux.data[userFieldRedux.data.length - 1].label
+      : '',
+  );
+  // use a state to control if currently is adding a point
+  const [isAddingPoint, setIsAddingPoint] = useState(true);
+  const [drawFields, setDrawFields] = useState([]);
+
+  const userFieldsRef = useRef(userFieldRedux ? [...userFieldRedux.data] : []);
+
+  const { isAuthenticated } = useAuth0();
+
+  // console.log('userFieldRedux', userFieldRedux);
+  // console.log('selectedUserField', selectedUserField)
+
+  const getArea = () => {
```

Comment:
why is this needed? the bit map already returns area right?

Reply:
This function sets the initFeatures for the map, whenever the user selected Fields changed, it sets the selected Field to the map.

Reply:
ok then can you rename it to something that makes more sense? like updateGeometry perhaps?